### PR TITLE
[Go] Abolish SSE

### PR DESCRIPTION
**Pull request by motoshi-suzuki-hp**
# Issue Number

https://github.com/okuda-seminar/Twitter-Clone/issues/840

# Implementation Summary

This work deprecates Server-Sent Events (SSE) and adopts a RESTful architecture. The transition replaces persistent event streams with stateless HTTP endpoints to standardize request–response interactions.

# Scope of Impact

By removing the requirement for long-lived connections, the proposed design improves horizontal scalability and operational robustness. Furthermore, it exhibits stronger alignment with a layered, service-oriented architectural pattern, thereby simplifying deployment, load balancing, and fault isolation.

# Particular points to check

- Consistency and completeness of the OpenAPI specifications reflecting REST endpoints.
- Full decommissioning of SSE pathways and correct substitution with REST interfaces, including edge cases and fallback behavior.

# Test

Execute the full test suite:

```
go test ./...
```

Successful execution should confirm interface compliance and regression integrity.

# Note

Modifications were also applied to components utilizing mutual exclusion (mutexes) and channels to ensure thread-safe behavior and compatibility with the RESTful design.

# Schedule

10/3

### CODE REVIEW

File: openapi/components/responses/get_reverse_chronological_home_timeline_response.yml [hunk 1]
```diff
@@ -1,38 +1,31 @@
...
+      properties:
+        UUID:
+          type: string
+        Valid:
+          type: boolean
+    text:
+      type: string
+    createdAt:
+      type: string
+      format: date-time
```
Comment (Amoonguses1) [inline openapi/components/responses/get_reverse_chronological_home_timeline_response.yml:31]: Let's use post.yml, repost.yml, and quote_repost.yml

**Review by Amoonguses1 — CHANGES_REQUESTED** [review]

**Comment by ryuju0911** [comment]
If this PR is merged into the main branch, the application will break because the frontend and iOS have not yet handled the update. You must avoid this situation.
Please create a new feature branch from the main branch and change the merge destination of this PR to that feature branch.

File: go/internal/application/usecase/api/update_notification.go [hunk 2]
```diff
@@ -4,9 +4,4 @@ import "x-clone-backend/internal/domain/entity"
 
 type UpdateNotificationUsecase interface {
```
[the remark this answers is not in the document]
Comment (motoshi-suzuki-hp) [reply go/internal/application/usecase/api/update_notification.go:5]: @Amoonguses1 

Please allow me to pose a question out of caution. The following files (four Usecase files and one constants file) will no longer be used, and I am considering removing them. In this situation, is it sufficient to delete them with the standard `rm` command, or should I remove them using `git rm` instead?

* `internal/application/usecase/api/update_notification.go`
* `internal/application/usecase/implementation/update_notification.go`
* `internal/application/usecase/fake/update_notification.go`
* `internal/application/usecase/injector/update_notification.go`
* `internal/domain/entity/timeline_event.go`

File: go/internal/controller/handler/get_reverse_chronological_home_timeline_test.go [hunk 3]
```diff
@@ -41,7 +36,7 @@ func TestGetReverseChronologicalHomeTimeline(t *testing.T) {
 			expectedCount: 1,
 		},
 		"get a target user and following users posts": {
-			setup: func(userAndFolloweePostsUsecase usecase.UserAndFolloweePostsUsecase, updateNotificationUsecase usecase.UpdateNotificationUsecase) {
+			setup: func(userAndFolloweePostsUsecase usecase.UserAndFolloweePostsUsecase) {
```
[the remark this answers is not in the document]
Comment (motoshi-suzuki-hp) [reply go/internal/controller/handler/get_reverse_chronological_home_timeline_test.go:40]: I added reposts and quote reposts in this test case named "get a target user and following users posts" .
